### DISCUSSION TITLE
Single file init.lua

### DISCUSSION THREAD

**Discussion by jackielii** [LazyVim/LazyVim#2199] [Ideas]
I'm porting my config to lazy.nvim. I quite like keeping everything one big file. A habit? easier to copy around? I just like it.

I didn't want to move directly to LazyVim since I thought it might be easier to copy and paste from LazyVim since it has so many nice setups that I sort of have but either slightly different or just less good.

As I copy and paste the sections I realised: Instead of using the lazyvim starter, can't I just adapt it to create an init.lua that just use lazy.nvim that imports LazyVim like the https://github.com/LazyVim/starter/blob/main/lua/config/lazy.lua? Has anybody done this?

**Comment by abeldekat** [comment]
Yes, it's possible.

I have done this just for fun: [Turning LazyVim into Kickstart](https://github.com/LazyVim/LazyVim/discussions/1483).
[Lazyflex.nvim](https://github.com/abeldekat/lazyflex.nvim) now contains some of the concepts used in this config.

However, the `LazyVim starter` structure provides the skeleton files for integrating your options, keymappings and autocommands with `LazyVim`'s setup. See directory `lua.config`.
If you want to be able to properly override `LazyVim`'s config, then I think you should consider the `starter` approach.
